src/mixins/nft.js

```diff
@@ -437,37 +439,37 @@ export default {
         ? ['/cosmos.nft.v1beta1.MsgSend', 'buy_nft', 'new_class']
         : ['/cosmos.nft.v1beta1.MsgSend', 'mint_nft', 'buy_nft', 'new_class'];
       const ignoreToList = this.nftIsWritingNFT ? LIKECOIN_NFT_API_WALLET : '';
-      const historyOnChain = await this.getNFTEventsAll({
+      const history = await this.getNFTEventsAll({
         actionType,
         ignoreToList,
       });
-      let history = historyOnChain;
+
+      const promises = [getPurchasePriceMap(this.$api, history)];
 
       if (this.nftIsWritingNFT) {
-        try {
-          const { data } = await this.$api.get(
-            getNFTHistory({
-              classId: this.classId,
-              nftId: this.nftId,
-            })
-          );
-          const historyInDB = data.list;
-          const eventMap = new Map();
-          historyOnChain.forEach(e => {
-            eventMap.set(`${e.txHash}-${e.nftId}-${e.event}`, e);
-          });
-          historyInDB.forEach(e => {
-            const key = `${e.txHash}-${e.nftId}-${e.event}`;
-            if (eventMap.has(key)) {
-              eventMap.get(key).price = e.price;
-            }
-          });
-          history = [...eventMap.values()];
-        } catch (error) {
-          // eslint-disable-next-line no-console
-          console.error(error);
-        }
+        promises.push(
+          getCollectPriceMap({
+            axios: this.$api,
+            classId: this.classId,
+            nftId: this.nftId,
+          })
+        );
       }
+
+      const eventMap = new Map();
+      history.forEach(event => {
+        eventMap.set(getEventKey(event), event);
+      });
+
+      const priceMaps = await Promise.all(promises);
+      priceMaps.forEach(priceMap => {
+        priceMap.forEach((price, key) => {
```

Comment:
So there is an hidden assumption that anything in `getCollectPriceMap` will overwrite `getCollectPriceMap` if they collide?

Reply:
Yeah..., could say so. While we don't have the purchase price in our Firestore yet. I didn't check if `price` will encounter collisions for now.